components/nexus-core/src/main/java/org/sonatype/nexus/proxy/storage/local/fs/DefaultFSLocalRepositoryStorage.java

```diff
@@ -333,6 +339,33 @@ public void storeItem(Repository repository, StorageItem item)
         prepareStorageFileItemForStore(fItem);
 
         cl = fItem.getContentLocator();
+
+        try {
+          final BufferedInputStream bufferedInputStream;
+          if (!cl.isReusable()) {
+            bufferedInputStream = new BufferedInputStream(cl.getContent());
+            bufferedInputStream.mark(1024);
+            // link persister with close the stream, so wrap it and prevent closing
+            cl = new PreparedContentLocator(new FilterInputStream(bufferedInputStream) {
+              @Override
+              public void close() {
+                // nop
+              }
```

Comment:
You could consider moving this anonymous subclass to the buffered stream, ie:
```
            bufferedInputStream = new BufferedInputStream(cl.getContent()) {
              @Override
              public void close() {
                // nop
              }
            };
```
which would avoid the need for the `FilterInputStream` and the extra layer of indirection.